### PR TITLE
Simplify return in script

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**
towards https://github.com/lfortran/lfortran/issues/3129

### CODE REVIEW

File: src/libasr/intrinsic_func_registry_util_gen.py [hunk 1]
```diff
@@ -340,7 +341,13 @@ def add_create_func_return_src(func_name):
...
     ret_type_val = arg_infos[0].get("return", None)
     ret_type_arg_idx = arg_infos[0].get("ret_type_arg_idx", None)
-    ret_type = ret_type_val if ret_type_val else f"expr_type(args[{ret_type_arg_idx}])"
+    if ret_type_val:
+        ret_type = ret_type_val
+    else:
+        src += indent * 2 + "ASRUtils::ExprStmtDuplicator expr_duplicator(al);\n"
+        src += indent * 2 + "expr_duplicator.allow_procedure_calls = true;\n"
+        src += indent * 2 + f"ASR::ttype_t* type_ = expr_duplicator.duplicate_ttype(expr_type(args[{ret_type_arg_idx}]));\n"
+        ret_type = "type_"
```
Comment (certik) [inline src/libasr/intrinsic_func_registry_util_gen.py:350]: Why does the expression have to be duplicated, and before it didn't?

**Review by certik — APPROVED** [review]
I think this looks good. Thanks for fixing it.

I left a question.

File: src/libasr/intrinsic_func_registry_util_gen.py (hunk 1 above)
Comment (Shaikh-Ubaid) [reply src/libasr/intrinsic_func_registry_util_gen.py:350]: We use the type from the arg as return type. Later we update the kind of the return type as per 2nd arg. This also undesirably updates the kind of the first argument whose type we took as return type.

Comment (Shaikh-Ubaid) [reply src/libasr/intrinsic_func_registry_util_gen.py:350]: We use the type from the arg as return type. Later we update the kind of the return type as per 2nd arg. This also undesirably updates the kind of the first argument whose type we took as return type.

Comment (certik) [reply src/libasr/intrinsic_func_registry_util_gen.py:350]: Ah, so there was a bug before?

**Comment by certik** [comment]
Please resolve conflicts, rebase and merge.

File: src/libasr/intrinsic_func_registry_util_gen.py (hunk 1 above)
Comment (Shaikh-Ubaid) [reply src/libasr/intrinsic_func_registry_util_gen.py:350]: Yes

Comment (Shaikh-Ubaid) [reply src/libasr/intrinsic_func_registry_util_gen.py:350]: There was also a bug in expr duplicator which is also fixed.

Comment (Shaikh-Ubaid) [reply src/libasr/intrinsic_func_registry_util_gen.py:350]: There was also a bug in expr duplicator which is also fixed.

Comment (certik) [reply src/libasr/intrinsic_func_registry_util_gen.py:350]: Perfect.

**Pull request merged**